CRM/Pledge/BAO/Pledge.php

```diff
@@ -1079,13 +1079,14 @@ public static function updatePledgeStatus($params) {
                   $activityType,
                   'name'
                 ),
-                'activity_date_time' => CRM_Utils_Date::isoToMysql($now),
+                'activity_date_time' => CRM_Utils_Date::isoToMysql(date('YmdHis')),
                 'due_date_time' => CRM_Utils_Date::isoToMysql($details['scheduled_date']),
                 'is_test' => $details['is_test'],
                 'status_id' => 2,
                 'campaign_id' => $details['campaign_id'],
               );
-              if (is_a(civicrm_api('activity', 'create', $activityParams), 'CRM_Core_Error')) {
+              $result = civicrm_api3('activity', 'create', $activityParams);
```

Comment:
calling civicrm_api3 will throw an exception on error - which can either be caught here or in the calling function